BlackrockSortingExtractor error when reading .nev file

When trying to use `BlackrockSortingExtractor` to read spikes from a `.nev` file:
```python
from spikeextractors import BlackrockSortingExtractor

se = BlackrockSortingExtractor(filename='file.nev')
```

This yields an error:
```
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-15-5c65512a0164> in <module>
      4 
      5 datafile = '/media/luiz/storage/taufferconsulting/client_ben/project_movshon/data/XX_LE_textures_20191128_002.nev'
----> 6 se = BlackrockSortingExtractor(filename=datafile)

/media/luiz/storage/Github/spikeextractors/spikeextractors/extractors/neoextractors/neobaseextractor.py in __init__(self, **kargs)
    150         # neo handle this but not spikeextractors
    151 
--> 152         self._handle_sampling_frequency()
    153 
    154     def _handle_sampling_frequency(self):

/media/luiz/storage/Github/spikeextractors/spikeextractors/extractors/neoextractors/neobaseextractor.py in _handle_sampling_frequency(self)
    160         #  all channels are in the same neo group so
    161         self._neo_sig_sampling_rate = self.neo_reader.header['signal_channels']['sampling_rate'][0]
--> 162         self._neo_sig_time_start = self.neo_reader.get_signal_t_start(self.block_index, self.seg_index,
    163                                                                       channel_indexes=None)
    164 

~/anaconda3/envs/env_movshon/lib/python3.8/site-packages/neo/rawio/baserawio.py in get_signal_t_start(self, block_index, seg_index, channel_indexes)
    460     def get_signal_t_start(self, block_index, seg_index, channel_indexes=None):
    461         if self._several_channel_groups:
--> 462             self._check_common_characteristics(channel_indexes)
    463         return self._get_signal_t_start(block_index, seg_index, channel_indexes)
    464 

~/anaconda3/envs/env_movshon/lib/python3.8/site-packages/neo/rawio/baserawio.py in _check_common_characteristics(self, channel_indexes)
    396         # ~ print('_check_common_characteristics', channel_indexes)
    397 
--> 398         assert channel_indexes is not None, \
    399             'You must specify channel_indexes'
    400         characteristics = self.header['signal_channels'][_common_sig_characteristics]

AssertionError: You must specify channel_indexes
```

--- a/spikeextractors/extractors/klustaextractors/klustaextractors.py
+++ b/spikeextractors/extractors/klustaextractors/klustaextractors.py
@@ -102,30 +102,34 @@ class KlustaSortingExtractor(SortingExtractor):
                 cs_to_exclude.append(ec.lower())
 
         for channel_group in kf_reader.get('/channel_groups'):
-            chan_cluster_id_arr = kf_reader.get(f'/channel_groups/{channel_group}/spikes/clusters/main')[()]
-            chan_cluster_times_arr = kf_reader.get(f'/channel_groups/{channel_group}/spikes/time_samples')[()]
-            chan_cluster_ids = np.unique(chan_cluster_id_arr)  # if clusters were merged in gui,
-                                                                # the original id's are still in the kwiktree, but
-                                                                # in this array
+            if 'spikes' not in kf_reader.get(f'/channel_groups/{channel_group}'):
+                print('No spikes found for this channel group')
+                continue
+            else:
+                chan_cluster_id_arr = kf_reader.get(f'/channel_groups/{channel_group}/spikes/clusters/main')[()]
+                chan_cluster_times_arr = kf_reader.get(f'/channel_groups/{channel_group}/spikes/time_samples')[()]
+                chan_cluster_ids = np.unique(chan_cluster_id_arr)  # if clusters were merged in gui,
+                                                                    # the original id's are still in the kwiktree, but
+                                                                    # in this array
 
-            for cluster_id in chan_cluster_ids:
-                cluster_frame_idx = np.nonzero(chan_cluster_id_arr == cluster_id)  # the [()] is a h5py thing
-                st = chan_cluster_times_arr[cluster_frame_idx]
-                assert st.shape[0] > 0, 'no spikes in cluster'
-                cluster_group = kf_reader.get(f'/channel_groups/{channel_group}/clusters/main/{cluster_id}').attrs['cluster_group']
+                for cluster_id in chan_cluster_ids:
+                    cluster_frame_idx = np.nonzero(chan_cluster_id_arr == cluster_id)  # the [()] is a h5py thing
+                    st = chan_cluster_times_arr[cluster_frame_idx]
+                    assert st.shape[0] > 0, 'no spikes in cluster'
+                    cluster_group = kf_reader.get(f'/channel_groups/{channel_group}/clusters/main/{cluster_id}').attrs['cluster_group']
 
-                assert cluster_group in self.default_cluster_groups.keys(), f'cluster_group not in "default_dict: {cluster_group}'
-                cluster_group_name = self.default_cluster_groups[cluster_group]
+                    assert cluster_group in self.default_cluster_groups.keys(), f'cluster_group not in "default_dict: {cluster_group}'
+                    cluster_group_name = self.default_cluster_groups[cluster_group]
 
-                if cluster_group_name.lower() in cs_to_exclude:
-                    continue
+                    if cluster_group_name.lower() in cs_to_exclude:
+                        continue
 
-                self._spiketrains.append(st)
-                klusta_units.append(int(cluster_id))
-                unique_units.append(unit)
-                unit += 1
-                groups.append(int(channel_group))
-                cluster_groups_name.append(cluster_group_name)
+                    self._spiketrains.append(st)
+                    klusta_units.append(int(cluster_id))
+                    unique_units.append(unit)
+                    unit += 1
+                    groups.append(int(channel_group))
+                    cluster_groups_name.append(cluster_group_name)
 
         if len(np.unique(klusta_units)) == len(np.unique(unique_units)):
             self._unit_ids = klusta_units

--- a/spikeextractors/extractors/neoextractors/blackrockextractor.py
+++ b/spikeextractors/extractors/neoextractors/blackrockextractor.py
@@ -57,6 +57,6 @@ class BlackrockSortingExtractor(NeoBaseSortingExtractor):
     NeoRawIOClass = 'BlackrockRawIO'
 
     def __init__(self, filename: PathType, nsx_to_load: Optional[int] = None,
-                 block_index: Optional[float] = None, seg_index: Optional[float] = None, **kwargs):
+                 block_index: Optional[int] = None, seg_index: Optional[int] = None, **kwargs):
         super().__init__(filename=filename, nsx_to_load=nsx_to_load,
                          block_index=block_index, seg_index=seg_index, **kwargs)

--- a/spikeextractors/extractors/neoextractors/blackrockextractor.py
+++ b/spikeextractors/extractors/neoextractors/blackrockextractor.py
@@ -31,8 +31,8 @@ class BlackrockRecordingExtractor(NeoBaseRecordingExtractor):
     installed = HAVE_NEO
     NeoRawIOClass = 'BlackrockRawIO'
 
-    def __init__(self, filename: PathType, nsx_to_load: Optional[int] = None, block_index: Optional[float] = None, 
-                 seg_index: Optional[float] = None, **kwargs):
+    def __init__(self, filename: PathType, nsx_to_load: Optional[int] = None, block_index: Optional[int] = None, 
+                 seg_index: Optional[int] = None, **kwargs):
         super().__init__(filename=filename, nsx_to_load=nsx_to_load, 
                          block_index=block_index, seg_index=seg_index, **kwargs)
 

--- a/spikeextractors/extractors/neoextractors/blackrockextractor.py
+++ b/spikeextractors/extractors/neoextractors/blackrockextractor.py
@@ -31,8 +31,10 @@ class BlackrockRecordingExtractor(NeoBaseRecordingExtractor):
     installed = HAVE_NEO
     NeoRawIOClass = 'BlackrockRawIO'
 
-    def __init__(self, filename: PathType, block_index: Optional[float] = None, seg_index: Optional[float] = None):
-        super().__init__(filename=filename, block_index=block_index, seg_index=seg_index)
+    def __init__(self, filename: PathType, nsx_to_load: Optional[int] = None, block_index: Optional[float] = None, 
+                 seg_index: Optional[float] = None, **kwargs):
+        super().__init__(filename=filename, nsx_to_load=nsx_to_load, 
+                         block_index=block_index, seg_index=seg_index, **kwargs)
 
 
 class BlackrockSortingExtractor(NeoBaseSortingExtractor):
@@ -53,3 +55,8 @@ class BlackrockSortingExtractor(NeoBaseSortingExtractor):
     mode = 'file'
     installed = HAVE_NEO
     NeoRawIOClass = 'BlackrockRawIO'
+
+    def __init__(self, filename: PathType, nsx_to_load: Optional[int] = None,
+                 block_index: Optional[float] = None, seg_index: Optional[float] = None, **kwargs):
+        super().__init__(filename=filename, nsx_to_load=nsx_to_load,
+                         block_index=block_index, seg_index=seg_index, **kwargs)

--- a/spikeextractors/extractors/neoextractors/neobaseextractor.py
+++ b/spikeextractors/extractors/neoextractors/neobaseextractor.py
@@ -50,7 +50,7 @@ class NeoBaseRecordingExtractor(RecordingExtractor, _NeoBaseExtractor):
 
     def __init__(self, block_index=None, seg_index=None, **kargs):
         RecordingExtractor.__init__(self)
-        _NeoBaseExtractor.__init__(self, block_index=None, seg_index=None, **kargs)
+        _NeoBaseExtractor.__init__(self, block_index=block_index, seg_index=seg_index, **kargs)
 
         # TODO propose a meachanisim to select the appropriate channel groups
         # in neo one channel group have the same dtype/sampling_rate/group_id
@@ -123,8 +123,8 @@ class NeoBaseRecordingExtractor(RecordingExtractor, _NeoBaseExtractor):
 
 
 class NeoBaseSortingExtractor(SortingExtractor, _NeoBaseExtractor):
-    def __init__(self, **kargs):
-        _NeoBaseExtractor.__init__(self, **kargs)
+    def __init__(self, block_index=None, seg_index=None, **kargs):
+        _NeoBaseExtractor.__init__(self, block_index=block_index, seg_index=seg_index, **kargs)
 
         # the sampling frequency is quite tricky because in neo
         # spike are handle in s or ms
@@ -146,7 +146,7 @@ class NeoBaseSortingExtractor(SortingExtractor, _NeoBaseExtractor):
         #  all channels are in the same neo group so
         self._neo_sig_sampling_rate = self.neo_reader.header['signal_channels']['sampling_rate'][0]
         self._neo_sig_time_start = self.neo_reader.get_signal_t_start(self.block_index, self.seg_index,
-                                                                      channel_indexes=None)
+                                                                      channel_indexes=[0])
 
         self.set_sampling_frequency(self._neo_sig_sampling_rate)
 

